contracts/polydex/Migrator/QuickSwapMigrator.sol

```diff
@@ -0,0 +1,255 @@
+    );
+
+    modifier ensureNonZeroAddress(address addressToCheck) {
+        require(
+            addressToCheck != address(0),
+            "QuickSwapMigrator: No zero address"
+        );
+        _;
+    }
+
+    constructor(
+        IPolydexRouter _polydexRouter,
+        IPolydexRouter _quickswapRouter,
+        IFarm _polydexFarm,
+        IDappFactoryFarm _quickswapFarm,
+        IRewardManager _rewardManger,
+        address _cnt
+    ) {
+        require(_cnt != address(0), "QuickSwapMigrator: No zero address");
+        polydexRouter = _polydexRouter;
+        quickswapRouter = _quickswapRouter;
+        polydexFactory = IPolydexFactory(_polydexRouter.factory());
+        quickswapFactory = IPolydexFactory(_quickswapRouter.factory());
+        wmatic = _polydexRouter.WETH();
+        polydexFarm = _polydexFarm;
+        quickswapFarm = _quickswapFarm;
+        rewardManager = _rewardManger;
+        cnt = _cnt;
+    }
+
+    // need to call addUserToWhiteList before this
+    //Prerequisite: in RewardManager excludedAddresses[LiquidityMigrator_Contract] & rewardDistributor[LiquidityMigrator_Contract] should be set to true
+    function migrate(
+        uint256 _oldPid,
+        uint256 _lpAmount,
+        IPolydexPair _oldLPAddress,
+        IPolydexPair _newLPAddress
+    )
+        external
+        nonReentrant
+        ensureNonZeroAddress(address(_oldLPAddress))
+        ensureNonZeroAddress(address(_newLPAddress))
+    {
+        //general checks
+        require(
+            _lpAmount > 0,
+            "QuickSwapMigrator: LP Amount should be greater than zero"
+        );
+        require(_oldPid == 0, "QuickSwapMigrator: Invalid pid");
+
+        //validate LP addresses
+        IPolydexPair oldLPAddress = IPolydexPair(
+            polydexFactory.getPair(wmatic, cnt)
+        );
+        IPolydexPair newLPAddress = IPolydexPair(
+            quickswapFactory.getPair(wmatic, cnt)
+        );
+        require(
+            _oldLPAddress == oldLPAddress && _newLPAddress == newLPAddress,
+            "QuickSwapMigrator: Invalid LP token addresses"
+        );
+
+        //Withdraw old LP tokens
+        polydexFarm.withdrawFor(_oldPid, _lpAmount, msg.sender);
+        require(
+            oldLPAddress.balanceOf(address(this)) >= _lpAmount,
+            "QuickSwapMigrator: Insufficient old LP Balance"
+        );
+
+        //Migrator vests users's CNT to reward manager for the user
+        uint256 cntBalance = IERC20(cnt).balanceOf(address(this));
+        if (cntBalance > 0) {
+            TransferHelper.safeTransfer(
+                address(cnt),
+                address(rewardManager),
+                cntBalance
+            );
+            rewardManager.handleRewardsForUser(
+                msg.sender,
+                cntBalance,
+                block.timestamp,
+                _oldPid,
+                0
+            );
+        }
+
+        liquidityVars.tokenA = oldLPAddress.token0();
```

Comment:
Can move the code from this line to `_transformLiquidity`. `_transformLiquidity` is only adding liquidity to quickswap atm.

Reply:
Sure will do that.